### PR TITLE
Observables

### PR DESCRIPTION
**Pull request by KiveshVirasamy**
Please check if my observable is okay

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    observer.next();
+    observer.complete();
+  }, (error) => {
+    observer.error(error);
+  });
+});
+
+// Observable to add event listener to the track flight button.
+const addEventListener$: Observable<any> = Observable.create((observer) => {
+  flightData$.subscribe((flightData) => {
```
Comment (CharelMarais) [inline src/main.ts:50]: You can combine 2 observables using `concatMap()` to get the output for the list of flights and then subscribe to that one observable on both `36` and `49`.

**Review by CharelMarais — COMMENTED** [review]
Looks good, I made one comment that you can review.

File: src/main.ts (hunk 1 above)
Comment (KiveshVirasamy) [reply src/main.ts:50]: Okay cool thanks man

Comment (KiveshVirasamy) [reply src/main.ts]: I added the concat map

File: src/main.ts [hunk 2]
```diff
@@ -1,68 +1,61 @@
...
   }
 });
 
-// Observable to add flight information to the container in the DOM.
-const addFlightInfo$: Observable<any> = Observable.create((observer) => {
-  flightData$.subscribe((flightData) => {
-    flightData.states.forEach((flight) => {
+const addFlightInfo$: Observable<void> = flightData$.pipe(
+  concatMap((flightData: FlightData) => {
+    flightData.states.forEach((flight: any) => {
```
Comment (MihlaliMgwebi) [inline src/main.ts:33]: Specifiy the type. Any is frowned upon.

File: src/utils/utilities.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+/**
+
+Calculates the heading direction from a given degree angle.
+@param {number} deg - The degree angle.
```
Comment (MihlaliMgwebi) [inline src/utils/utilities.ts:4]: Great comments

**Review by MihlaliMgwebi — CHANGES_REQUESTED** [review]
Can approve after change.

File: src/main.ts (hunk 2 above)
Comment (KiveshVirasamy) [reply src/main.ts:33]: okay on it

File: src/utils/utilities.ts (hunk 3 above)
Comment (KiveshVirasamy) [reply src/utils/utilities.ts:4]: Thanks

**Review by MihlaliMgwebi — APPROVED** [review]

**Pull request merged**